src/hotspot/cpu/s390/frame_s390.hpp

```diff
@@ -330,6 +330,10 @@
 #define _z_ijava_state_neg(_component) \
          (int) (-frame::z_ijava_state_size + offset_of(frame::z_ijava_state, _component))
 
+// Frame slot index relative to fp
+#define _z_ijava_idx(_component) \
+        (_z_ijava_state_neg(_component) >> LogBytesPerWord)
+
```

Comment:
Should this be a function?
Also, names starting with `_` aren't common in HotSpot code, except for fields in C++ objects.

Reply:
> Should this be a function?

This is similar to [ppc ijava_idx](https://github.com/openjdk/jdk/blob/885be2efa6b1359a7c7ab36882e19a7eaba77fb3/src/hotspot/cpu/ppc/frame_ppc.hpp#L283C1-L285C58)

> Also, names starting with _ aren't common in HotSpot code, except for fields in C++ objects.

I did it for keeping the parity with: 
```cpp
#define _z_ijava_state_neg(_component) \
         (int) (-frame::z_ijava_state_size + offset_of(frame::z_ijava_state, _component))
```
Do you want me to revert it ?

Reply:
There is no good reason to use a macro here. If a function can be a function, and this one can, let it be one. However, there is no reason to change anything else. Leave that for a "macros to functions" patch some other day.

Reply:
I could only achieve this implementation: [macro to method](https://github.com/offamitkumar/jdk/commit/a4a3908288c8c0f518ad263062acd504cd6b7d3c), which looks dirty.

Reply:
Fair enough. It's not worth doing that, and what you do here does match PPC. OK.